### PR TITLE
Feature/reportng renewed threads view

### PR DESCRIPTION
**Pull request by TobiasAdler**
# Description

The old threads view was replaced with a new one using Apache ECharts.
Details of a testcase (method name, run index, start- and end-time and duration) are provided with a tooltip.
In Addition to the method filter, a status filter was added. With the selection of a status all testcases of that status are highlighted and zoomed in on. These two filters can not be active at the same time, instead they cancel each other out.
The Method filter itself highlights the selected testcase and performs a zoom event on that specific method.

New Threads View with active tooltip:
threads_new_2

Status Filter active:
threads_new_4

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: report-ng/app/src/app.ts [hunk 1]
```diff
@@ -80,12 +80,6 @@ export class App {
                 name: "tests",
                 title: 'Tests'
             },
-            // {
-            //     route: 'threads',
```
Comment (mreiche) [inline report-ng/app/src/app.ts:84]: How do you reach the Thread view now?

File: report-ng/app/package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     ],
     "license": "Apache-2.0",
     "dependencies": {
+        "@aurelia-mdc-web/button": "^7.1.3",
```
Comment (mreiche) [inline report-ng/app/package.json:12]: Button has been added, but not used in `threads.html`

File: report-ng/app/src/styles/mdc.scss [hunk 3]
```diff
@@ -25,7 +25,7 @@
   $on-primary: #ffffff,
   $on-secondary: #ffffff,
 );
-//@use '@material/button/mdc-button';
+@use '@material/button/mdc-button';
```
Comment (mreiche) [inline report-ng/app/src/styles/mdc.scss:28]: Same here. No `mdc-button` used anywhere.

File: report-ng/app/webpack.config.js [hunk 4]
```diff
@@ -37,7 +37,8 @@ module.exports = ({production} = {}, {analyze, tests, hmr, port, host} = {}) =>
                 // https://github.com/aurelia/binding/issues/702
                 // Enforce single aurelia-binding, to avoid v1/v2 duplication due to
                 // out-of-date dependencies on 3rd party aurelia plugins
-                'aurelia-binding': path.resolve(__dirname, 'node_modules/aurelia-binding')
+                'aurelia-binding': path.resolve(__dirname, 'node_modules/aurelia-binding'),
+                'tslib': path.resolve(__dirname, 'node_modules/tslib')
```
Comment (mreiche) [inline report-ng/app/webpack.config.js:41]: What solves this?

**Review by mreiche — CHANGES_REQUESTED** [review]

File: report-ng/app/package.json [hunk 5]
```diff
@@ -39,10 +40,13 @@
         "aurelia-templating": "^1.10.4",
         "aurelia-typed-observable-plugin": "^0.4.2",
         "aurelia-ui-virtualization": "^1.0.0-rc2",
+        "echarts": "5.4.2",
         "highlight.js": "^11.8.0",
         "moment": "^2.29.4",
         "moment-duration-format": "^2.3.2",
-        "t-systems-aurelia-components": "^1.1.0",
+        "pixelmatch": "^5.2.1",
+        "protobufjs": "^6.11.3",
```
Comment (mreiche) [inline report-ng/app/package.json:48]: `pixelmatch` and `protobufjs` have been added and not updated?

File: report-ng/app/src/main.ts [hunk 6]
```diff
@@ -32,6 +32,7 @@ export function configure(aurelia: Aurelia) {
     aurelia.use
         .standardConfiguration()
         .plugin(PLATFORM.moduleName('@aurelia-mdc-web/base'))
+        .plugin(PLATFORM.moduleName('@aurelia-mdc-web/button'))
```
Comment (mreiche) [inline report-ng/app/src/main.ts:35]: Not used `mdc-button`

File: report-ng/app/src/app.ts (hunk 1 above)
Comment (TobiasAdler) [reply report-ng/app/src/app.ts:84]: The Thread view is reached in the exact same way as before (line 118). That was just a duplicated code fragment I removed.

File: report-ng/app/webpack.config.js (hunk 4 above)
Comment (TobiasAdler) [reply report-ng/app/webpack.config.js:41]: The Apache ECharts are using a different version of `tslib`.
(In this case: v2.3.0 from echarts and v2.3.1 from mdc-component)
This caused a duplicate-package-warning, which was resolved with the extra line
`'tslib': path.resolve(__dirname, 'node_modules/tslib')`.

File: report-ng/app/package.json (hunk 5 above)
Comment (TobiasAdler) [reply report-ng/app/package.json]: `pixelmatch` and `protobufjs` are now updated.

File: report-ng/app/src/main.ts (hunk 6 above)
Comment (TobiasAdler) [reply report-ng/app/src/main.ts]: `mdc-button` fragments have now been removed.

File: report-ng/app/src/styles/mdc.scss (hunk 3 above)
Comment (TobiasAdler) [reply report-ng/app/src/styles/mdc.scss]: `mdc-button` fragments have now been removed.

File: report-ng/app/package.json (hunk 2 above)
Comment (TobiasAdler) [reply report-ng/app/package.json]: `mdc-button` fragments have now been removed.

File: report-ng/app/webpack.config.js (hunk 4 above)
Comment (mreiche) [reply report-ng/app/webpack.config.js:41]: You should add this info at least as a comment above or behind this line.

File: report-ng/app/package.json (hunk 5 above)
Comment (mreiche) [reply report-ng/app/package.json]: No I mean, where does this dependency comes from in the first place?
`pixelmatch` and `protobufjs` aren't used before, why are they now?

Comment (TobiasAdler) [reply report-ng/app/package.json]: You are right. These dependencies were added with a merge some time ago and are indeed no longer required. They have now been removed.

File: report-ng/app/webpack.config.js (hunk 4 above)
Comment (TobiasAdler) [reply report-ng/app/webpack.config.js:43]: A comment was added.

**Review by mreiche — DISMISSED** [review]

**Comment by martingrossmann** [comment]
In case of many configuration methods (e.g. many `BeforeMethods`) the method list contains many entries with the same name:
image

Please add the run index in brackets like `setUp (20)` to increase clarity.

**Comment by martingrossmann** [comment]
@TobiasAdler Please also update the ReportNG tests. We have some failed tests because of changed Threads view.

**Review by mreiche — DISMISSED** [review]

**Review by mbeuthan — APPROVED** [review]

**Pull request merged**